app/models/user.rb

```diff
@@ -9,4 +9,27 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friends, through: :friendship
+  has_many :friends, through: :friendships_confirmation
+  has_many :friend_requests, through: :reverse_friendships, source: :user
```

Comment:
- [ ] The friendship feature has not been fully implemented. Currently, a user can press the Add friend button but nothing happens except leading to an empty page. A user should be able to accept or reject friend requests.
- [ ] There should be a button to add friend on user's single page.
- [ ] A user should be able to see all friend requests sent their way.